### PR TITLE
feat(lb): Add cross-region internal application load balancer

### PR DESCRIPTION
**Pull request by sahanacp-hub**
Intending to use this snippet on the following page:
https://cloud.google.com//load-balancing/docs/l7-internal/setting-up-l7-cross-reg-internal

### CODE REVIEW

File: lb/cross_reg_int_app_lb_mig_backend/main.tf [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+      # add external ip
+    }
+  }
+  boot_disk {
+    initialize_params {
+      image = "debian-cloud/debian-11"
+    }
+  }
+  # Without this, terraform thinks the image changes each time because
+  # bct-staging-images/debian-11 exands to
```
Comment (msampathkumar) [inline lb/cross_reg_int_app_lb_mig_backend/main.tf:375]: Help me understand why this comment, when image `bct-staging-images/debian-11` is not being used here!

File: lb/cross_reg_int_app_lb_mig_backend/main.tf [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or 
+implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+# Cross-region internal application load balancer with managed instance 
+group backend
```
Comment (msampathkumar) [inline lb/cross_reg_int_app_lb_mig_backend/main.tf:19]: Fix this line please!

File: lb/cross_reg_int_app_lb_mig_backend/main.tf [hunk 3]
```diff
@@ -0,0 +1,384 @@
...
+resource "google_compute_subnetwork" "subnet_b" {
+  provider      = google-beta
+  ip_cidr_range = "10.1.3.0/24"
+  name          = "lbsubnet-useast1"
+  network       = google_compute_network.default.id
+  region        = "us-east1"
+}
+# [END cloudloadbalancing_cross_region_subnet_b_tag]
+
+# proxy-only-subnet in us-east1
```
Comment (msampathkumar) [inline lb/cross_reg_int_app_lb_mig_backend/main.tf:53]:
```suggestion
# proxyonly-subnet in us-east1
```

File: lb/cross_reg_int_app_lb_mig_backend/main.tf [hunk 4]
```diff
@@ -0,0 +1,384 @@
...
+resource "google_compute_backend_service" "default" {
+  name                  = "gl7-gilb-backend-service"
+  provider              = google-beta
+  protocol              = "HTTP"
+  load_balancing_scheme = "INTERNAL_MANAGED"
+  timeout_sec           = 10
+  health_checks         = [google_compute_health_check.default.id]
+  backend {
+    group           = 
+google_compute_region_instance_group_manager.mig_a.instance_group
```
Comment (msampathkumar) [inline lb/cross_reg_int_app_lb_mig_backend/main.tf:146]: This line may need lint formatting here.

File: lb/cross_reg_int_app_lb_mig_backend/main.tf [hunk 5]
```diff
@@ -0,0 +1,384 @@
...
+    balancing_mode  = "UTILIZATION"
+    capacity_scaler = 1.0
+  }
+  backend {
+    group           = 
+google_compute_region_instance_group_manager.mig_b.instance_group
```
Comment (msampathkumar) [inline lb/cross_reg_int_app_lb_mig_backend/main.tf:152]: This line may need lint formatting here.

File: lb/cross_reg_int_app_lb_mig_backend/main.tf [hunk 6]
```diff
@@ -0,0 +1,384 @@
...
+  }
+  boot_disk {
+    initialize_params {
+      image = "debian-cloud/debian-11"
+    }
+  }
+  # Without this, terraform thinks the image changes each time because
+  # bct-staging-images/debian-11 exands to
+  # debian-11--basic-gce-staging-bct-20230814-2318. Terraform thinks this 
+is a
```
Comment (msampathkumar) [inline lb/cross_reg_int_app_lb_mig_backend/main.tf:377]: Fix this line! Not terraform code.

File: lb/cross_reg_int_app_lb_mig_backend/main.tf (hunk 1 above)
Comment (sahanacp-hub) [reply lb/cross_reg_int_app_lb_mig_backend/main.tf:375]: This is a test-vm and the code was copied from a staging TF script. I have removed test-vm from the script.

File: lb/cross_reg_int_app_lb_mig_backend/main.tf (hunk 2 above)
Comment (sahanacp-hub) [reply lb/cross_reg_int_app_lb_mig_backend/main.tf:19]: Fixed

File: lb/cross_reg_int_app_lb_mig_backend/main.tf (hunk 3 above)
Comment (sahanacp-hub) [reply lb/cross_reg_int_app_lb_mig_backend/main.tf:53]: Fixed

File: lb/cross_reg_int_app_lb_mig_backend/main.tf [hunk 7]
```diff
@@ -0,0 +1,384 @@
...
+  purpose       = "GLOBAL_MANAGED_PROXY"
+  region        = "us-east1"
+  role          = "ACTIVE"
+  lifecycle {
+    ignore_changes = [ipv6_access_type]
+  }
+}
+# [END cloudloadbalancing_cross_region_proxy_subnet_b_tag]
+
+# proxy-only-subnet in us-west1
```
[the remark this answers is not in the document]
Comment (sahanacp-hub) [reply lb/cross_reg_int_app_lb_mig_backend/main.tf:69]: Fixed

File: lb/cross_reg_int_app_lb_mig_backend/main.tf (hunk 6 above)
Comment (sahanacp-hub) [reply lb/cross_reg_int_app_lb_mig_backend/main.tf:377]: fixed

File: lb/cross_reg_int_app_lb_mig_backend/main.tf (hunk 4 above)
Comment (sahanacp-hub) [reply lb/cross_reg_int_app_lb_mig_backend/main.tf:134]: We cannot fit this line in the 80 character length limit. Kindly ignore this error.

Comment (sahanacp-hub) [reply lb/cross_reg_int_app_lb_mig_backend/main.tf:134]: We cannot fit this line in the 80 character length limit. Kindly ignore this error.

File: lb/cross_reg_int_app_lb_mig_backend/main.tf (hunk 5 above)
Comment (sahanacp-hub) [reply lb/cross_reg_int_app_lb_mig_backend/main.tf:140]: we cannot fit the variable in the 80 character limit. kindly ignore this error

**Comment by sahanacp-hub** [comment]
The 80 character limit cannot be met for some values. Kindly ignore the errors

**Comment by sahanacp-hub** [comment]
Working on fixing the lint issues

**Review by msampathkumar — APPROVED** [review]
LGTM

**Review by gericdong — APPROVED** [review]

**Pull request merged**